### PR TITLE
EOS-19624 Password is exposed during add node

### PR DESCRIPTION
**Pull request by akash2144**
Signed-off-by: Akash Dudhane <akash.a.dudhane@seagate.com>

## Problem Statement
<pre>
  <code>
  Story Ref (if any): EOS-19624
    Password is exposed during add node
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes/No
  </code>
</pre>
## Problem Description
<pre>
  <code>
    We are passing command as it is to log file without masking password during add node.
  </code>
</pre>
## Solution
<pre>
  <code>
    Here we are creating copy of command then password is masked in that copied command and passed to log fie.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Unit Testing details here...
  </code>
</pre>

### CODE REVIEW

**Review by ajay-paratmandali — APPROVED** [review]

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: ha/remote_execution/ssh_communicator.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(self, hostname, username=None, port=22, password=None):
             Log.error(f'SSHRemoteExecutor, some error occured while connecting \
                         to SSH channel {err}')
 
-    def execute(self, command: str) -> None:
+    def execute(self, command: str, secret=None) -> None:
```
Comment (ajay-paratmandali) [inline ha/remote_execution/ssh_communicator.py:44]: Please check RemoteExecutor parent class of current class and update parent class as 
overridden method should have same definition

Comment (akash2144) [reply ha/remote_execution/ssh_communicator.py:44]: RemoteExecutor's execute method updated.

File: ha/remote_execution/remote_executor.py [hunk 2]
```diff
@@ -30,6 +30,6 @@ def __init__(self, hostname, port):
         self._port = port
 
     @abc.abstractmethod
-    def execute(self, command):
+    def execute(self, command, secret=None):
```
Comment (indrajitzagade) [inline ha/remote_execution/remote_executor.py:33]: Why this change?

**Comment by indrajitzagade** [comment]
@akash2144 Please fix the codacy issues.

File: ha/remote_execution/remote_executor.py (hunk 2 above)
Comment (ajay-paratmandali) [reply ha/remote_execution/remote_executor.py:33]: Akash keep defination same on both class
```
def execute(self, command: str, secret=None) -> None:
def execute(self, command, secret=None):
```
Keep it
def execute(self, command: str, secret: str=None):

Comment (akash2144) [reply ha/remote_execution/remote_executor.py:33]: This is RemoteExecutor parent class's execute method. For overriding, function must have same defination.

Comment (akash2144) [reply ha/remote_execution/remote_executor.py]: > Akash keep defination same on both class
> 
> ```
> def execute(self, command: str, secret=None) -> None:
> def execute(self, command, secret=None):
> ```
> 
> Keep it
> def execute(self, command: str, secret: str=None):

Function defination changed to : 
def execute(self, command: str, secret: str=None):

Comment (indrajitzagade) [reply ha/remote_execution/remote_executor.py]: Why we need this change @ajay-paratmandali ?

**Review by ajaysrivas — APPROVED** [review]

File: ha/remote_execution/remote_executor.py (hunk 2 above)
Comment (ajay-paratmandali) [reply ha/remote_execution/remote_executor.py]: We are overriding `execute` method from parent class 
In overriding method should have same parameter.

If we keep different in parent and child class then we will get error
There is also warning from codacy
```
Parameters differ from overridden 'execute' method
```
as Execute method parameter different in parent and child class

**Review by indrajitzagade — APPROVED** [review]
